File: src/DentalDrill.CRM/Models/InventoryMovementType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DentalDrill.CRM.Models
{
    public enum InventoryMovementType
    {
        /// <summary>Movement that sets the initial quantity as part of the initial stock data import.</summary>
        Initial = 0,

        /// <summary>Movement represents an order.</summary>
        Order = 1,

        /// <summary>Movement that corrects quantity by increasing it.</summary>
        Found = 2,

        /// <summary>Movement that transfer parts from another workshop.</summary>
        MoveFromAnotherWorkshop = 100,

        EphemeralMissingRequiredQuantity = 500,

        /// <summary>Movement that is created as part of repair process and would result in SKU consumption.</summary>
        Repair = 1000,

        /// <summary>Movement that corrects quantity by decreasing it.</summary>
        Lost = 1001,

        /// <summary>Movement that is created as part of repair process in case of partial allocation.</summary>
        RepairFragment = 1002,

        /// <summary>Movement that transfer parts to another workshop.</summary>
        MoveToAnotherWorkshop = 1100,
    }
}

File: src/DentalDrill.CRM/wwwroot/js/app/pages/inventoryMovements/shared/InventoryMovementType.ts
namespace DentalDrill.CRM.Pages.InventoryMovements.Shared {
    export enum InventoryMovementType {
        Initial = 0,
        Order = 1,
        Found = 2,
        MoveFromAnotherWorkshop = 100,
        EphemeralMissingRequiredQuantity = 500,
        Repair = 1000,
        Lost = 1001,
        RepairFragment = 1002,
        MoveToAnotherWorkshop = 1100,
    }

    export enum InventoryMovementCreateType {
        Order = 1,
        Found = 2,
        Lost = 1001,
        MoveBetweenWorkshops = 5000,
        MoveFromAnotherWorkshop = 100,
        MoveToAnotherWorkshop = 1100,
    }

    export class InventoryMovementTypeHelper {
        static toDisplayString(value: InventoryMovementType): string {
            switch (value) {
                case InventoryMovementType.Initial:
                    return "Initial";
                case InventoryMovementType.Order:
                    return "Order";
                case InventoryMovementType.Found:
                    return "Found";
                case InventoryMovementType.MoveFromAnotherWorkshop:
                    return "Move from another workshop";
                case InventoryMovementType.EphemeralMissingRequiredQuantity:
                    return "Pending Order";
                case InventoryMovementType.Repair:
                    return "Repair";
                case InventoryMovementType.Lost:
                    return "Lost";
                case InventoryMovementType.RepairFragment:
                    return "Repair (Partial)";
                case InventoryMovementType.MoveToAnotherWorkshop:
                    return "Move to another workshop";
                default:
                    throw new Error("Invalid movement type");
            }
        }

        static createDataSource(): kendo.data.DataSource {
            const items: InventoryMovementType[] = [
                InventoryMovementType.Initial,
                InventoryMovementType.Order,
                InventoryMovementType.Found,
                InventoryMovementType.MoveFromAnotherWorkshop,
                InventoryMovementType.EphemeralMissingRequiredQuantity,
                InventoryMovementType.Repair,
                InventoryMovementType.Lost,
                InventoryMovementType.RepairFragment,
                InventoryMovementType.MoveToAnotherWorkshop,
            ];

            return new kendo.data.DataSource({
                data: items.map(x => ({ name: InventoryMovementTypeHelper.toDisplayString(x), value: x }))
            });
        }
    }

    export class InventoryMovementCreateTypeHelper {
        static toDisplayString(value: InventoryMovementCreateType): string {
            switch (value) {
                case InventoryMovementCreateType.Order:
                    return "Order";
                case InventoryMovementCreateType.Found:
                    return "Found";
                case InventoryMovementCreateType.Lost:
                    return "Lost";
                case InventoryMovementCreateType.MoveBetweenWorkshops:
                    return "Move between workshops";
                case InventoryMovementCreateType.MoveFromAnotherWorkshop:
                    return "Move from another workshop";
                case InventoryMovementCreateType.MoveToAnotherWorkshop:
                    return "Move to another workshop";
                default:
                    throw new Error("Invalid movement type");
            }
        }
    }
}